### PR TITLE
Allow for dynamic terminal width inside AppStatus

### PR DESCRIPTION
**Pull request by nemanja-boric-sociomantic**
AppStatus was assuming that main terminal exists and that
it can be queried for its width. However this is not true
when starting programs as daemons from systemd - the main terminal
doesn't exist (stdout may or may be not redirected), so the
width is zero. However, user may want to attach to the unix socket
and get the output from the AppStatus.

AppStatus is changed so it a) accept assumed terminal width at
construction which, b) can be overridden by passing the width
to the command (`app_status 120` for example).

### CODE REVIEW

**Comment by don-clugston-sociomantic** [comment]
Why not make the default width of 80 columns, rather than 0?

**Comment by nemanja-boric-sociomantic** [comment]
It's a bit confusing and I don't have a good answer. The default terminal width is whatever reported by the terminal. In case that's 0, we can make this value ` = 80` and that would override the zero. But what if the width of the terminal is 120, and this makes it drop to 80? I think its clearer if this value overrides the defaults, no matter if they are zero or not.

Alternatively, this could be renamed to `fallback_terminal_width` and set to 80.

**Review by mihails-strasuns — APPROVED** [review]

File: src/ocean/io/console/AppStatus.d [hunk 1]
```diff
@@ -445,6 +471,13 @@ public class AppStatus
         this.connectOutput(unix_socket_file,
             unix_terminal_output);
 
+        if (command.length > 0)
+        {
+            int columns;
+            toInteger(command[0], columns);
```
Comment (daniel-zullo) [inline src/ocean/io/console/AppStatus.d]: Optional: `this.terminal_columns` can be directly used here instead.

File: src/ocean/io/console/AppStatus.d [hunk 2]
```diff
@@ -359,6 +374,16 @@ public class AppStatus
         this.stream = stream;
         this.terminal_output = terminal_output;
         this.old_terminal_size = Terminal.rows;
+
+        if (terminal_columns)
```
Comment (daniel-zullo) [inline src/ocean/io/console/AppStatus.d]: you might want to explicitly compare `terminal_columns > 0`

**Comment by don-clugston-sociomantic** [comment]
> It's a bit confusing and I don't have a good answer. The default terminal width is whatever reported by the terminal. In case that's 0, we can make this value = 80 and that would override the zero. But what if the width of the terminal is 120, and this makes it drop to 80? I think its clearer if this value overrides the defaults, no matter if they are zero or not.

The point is that the terminal width should never be zero under any circumstances. A zero width terminal is nonsense, it's not a terminal. I guess Linux allows it to avoid a singularity in the API but we're high enough level of abstraction that we can remove it.

**Comment by nemanja-boric-sociomantic** [comment]
Updated based on both @daniel-zullo-sociomantic and @don-clugston-sociomantic 's suggestions.

**Comment by don-clugston-sociomantic** [comment]
LGTM

**Pull request merged**